honeycomb-core/src/spatial_repr/coords.rs

```diff
@@ -215,28 +215,40 @@ impl<T: CoordsFloat> IndexMut<usize> for Coords2<T> {
 #[cfg(test)]
 mod tests {
     use super::*;
-    use crate::common::FloatType;
-
-    fn almost_equal(lhs: &Coords2<FloatType>, rhs: &Coords2<FloatType>) -> bool {
-        const EPS: FloatType = 10.0e-12;
-        ((lhs.x - rhs.x).abs() < EPS) & ((lhs.y - rhs.y).abs() < EPS)
-    }
-
-    #[test]
-    fn sum() {
-        let collection = [
-            Coords2::unit_x(),
-            Coords2::unit_x(),
-            Coords2::unit_x(),
-            Coords2::unit_y(),
-            Coords2::unit_y(),
-            Coords2::unit_y(),
-        ];
-
-        let owned_sum: Coords2<FloatType> = collection.into_iter().sum();
-        let borrowed_sum: Coords2<FloatType> = collection.iter().sum();
-        let ref_value: Coords2<FloatType> = Coords2::from((3.0, 3.0));
-        assert!(almost_equal(&owned_sum, &ref_value));
-        assert!(almost_equal(&borrowed_sum, &ref_value));
-    }
+
+    macro_rules! almost_equal {
+        ($f1: expr, $f2: expr, $t: ty) => {
+            ($f1 - $f2).abs() < <$t>::EPSILON
+        };
+    }
```

Comment:
Not related to this PR, but it is not very correct.

Floating point numbers are designed to work with relative error, not absolute. So if you deal with large or small numbers, if comparison does not mean the same thing.

Reply:
I found this [reference](https://floating-point-gui.de/errors/comparison/) for comparing floating point numbers. If it looks good to you, I'll implement something similar later since I plan to group all tests of the `spatial_repr` module.